### PR TITLE
fix/services/php8/orgunits

### PR DESCRIPTION
**Pull request by mstuder**

### CODE REVIEW

**Comment by mstuder** [comment]
@see https://github.com/ILIAS-eLearning/ILIAS/pull/4106

**Comment by mstuder** [comment]
@mjansenDatabay sorry my question - i'm not sure it some one has to review this or if I may commit it without reviewing.

File: Services/UIComponent/Explorer2/classes/class.ilExplorerBaseGUI.php [hunk 1]
```diff
@@ -179,7 +179,7 @@ abstract public function getNodeId($a_node);
      * @param object|array $a_node
      * @return string href attribute
      */
-    public function getNodeHref($a_node) : string
+    public function getNodeHref(object|array $a_node) : string
```
Comment (mjansenDatabay) [inline Services/UIComponent/Explorer2/classes/class.ilExplorerBaseGUI.php:182]: @mstuder We also have to support PHP 7.4, so `Union Types` must not be used.

File: Services/UIComponent/Explorer2/classes/class.ilExplorerBaseGUI.php [hunk 2]
```diff
@@ -160,7 +160,7 @@ abstract public function getChildsOfNode($a_parent_node_id) : array;
      * @param object|array $a_node node array or object
      * @return string content of the node
      */
-    abstract public function getNodeContent($a_node) : string;
+    abstract public function getNodeContent(object|array $a_node) : string;
```
Comment (mjansenDatabay) [inline Services/UIComponent/Explorer2/classes/class.ilExplorerBaseGUI.php:163]: Unfortunately we must not use PHP 8 features in ILIAS 8.

**Comment by mjansenDatabay** [comment]
@mstuder We decided ...

- ... that there will be an organized code reviews (by other maintainers/Service Providers) after a component has been declared as `PHP 8 Compatibility Achieved` by the respective component maintainer.
- ... that simple/trivial fixes in the code of other maintainers can be committed/pushed without a PR/review, but the corresponding maintainers should be informed (e.g. by mentioning them here on GitHub).
- ... that all CI builds/unit tests must be passed before code is integrated into `trunk`.

Since you'll pass over the maintenance (of this component) to @klees after the PHP 8 project I also recommend to keep him posted regarding the current refactoring progress.

In your commits I see some usages of features that require PHP 8. Please keep in mind that we have to support PHP 7.4 as well.

**Comment by mstuder** [comment]
Thank you @mjansenDatabay 

Please excuse my lack of information at this point. I close this pullrequest and the orgunit pullrequest https://github.com/ILIAS-eLearning/ILIAS/pull/4106 now. I will make the development php7.4 compatible together with the orgunit developments.

I will then take @klees into the flow of information.

Best regards
Martin

**Comment by mjansenDatabay** [comment]
> Please excuse my lack of information at this point.

No problem.

**Pull request closed without merging**